src/controller/artifact/processor/default.go

```diff
@@ -44,16 +55,113 @@ func (d *defaultProcessor) GetArtifactType() string {
 	// can not get the artifact type from the media type, return unknown
 	return ArtifactTypeUnknown
 }
-func (d *defaultProcessor) ListAdditionTypes() []string {
-	return nil
+func (d *defaultProcessor) ListAdditionTypes(ctx context.Context, artifact *artifact.Artifact) []string {
+	artifactConfig := artifact.ExtraAttrs
+	schemaMap, ok := artifactConfig[schemaV1.XHarborAttributes]
+	if !ok {
+		return nil
+	}
+	schema := &schemaV1.Schema{}
+	mapstructure.Decode(schemaMap, schema)
+	additionTypes := make([]string, 0)
+	for _, addition := range schema.Additions {
+		additionTypes = append(additionTypes, addition.Name)
+	}
+	return additionTypes
 }
-func (d *defaultProcessor) AbstractMetadata(ctx context.Context, manifest []byte, artifact *artifact.Artifact) error {
-	// do nothing currently
-	// we can extend this function to abstract the metadata in the future if needed
+func (d *defaultProcessor) AbstractMetadata(ctx context.Context, artifact *artifact.Artifact, manifest []byte) error {
+	if artifact.ManifestMediaType != v1.MediaTypeImageManifest {
```

Comment:
Add support for docker's manifest media type: https://github.com/goharbor/harbor/blob/master/src/controller/artifact/abstractor.go#L64

Reply:
+1